lib/types/index.d.ts

```diff
@@ -765,6 +768,46 @@ export namespace Rule {
         hasSuggestions?: boolean | undefined;
     }
 
+    /** Provides additional metadata about a deprecation. */
+    interface DeprecatedInfo {
```

Comment:
The types `DeprecatedInfo`, `ReplacedByInfo` and `ExternalSpecifier` are now included in `@eslint/core` and can be imported from there.